tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -356,6 +356,39 @@ bool DeclResultIdMapper::createStageVars(const DeclaratorDecl *decl,
     if (varId == 0)
       return false;
 
+    // Decorate with interpolation modes for pixel shader input variables
+    if (shaderModel.IsPS() && sigPoint->IsInput()) {
+      const QualType elemType = typeTranslator.getElementType(type);
+
+      if (elemType->isBooleanType() || elemType->isIntegerType()) {
+        // TODO: Probably we can call hlsl::ValidateSignatureElement() for the
+        // following check.
+        if (decl->getAttr<HLSLLinearAttr>() ||
+            decl->getAttr<HLSLCentroidAttr>() ||
+            decl->getAttr<HLSLNoPerspectiveAttr>() ||
+            decl->getAttr<HLSLSampleAttr>()) {
+          emitError("only nointerpolation mode allowed for integer input "
```

Comment:
there's a discrepancy between the error message and the if condition.
it says "for integer input parameters", but the condition also checks for booleans.